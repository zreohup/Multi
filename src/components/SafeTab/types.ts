export interface safeTabItem<T> {
  label: string
  Component: React.FC<T>
}
