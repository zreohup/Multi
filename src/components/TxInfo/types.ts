import { Transaction } from '@safe-global/store/gateway/AUTO_GENERATED/transactions'
import { TxType } from '@/src/hooks/useTransactionType'

export type TxCardPress = { tx: Transaction; type?: TxType }
