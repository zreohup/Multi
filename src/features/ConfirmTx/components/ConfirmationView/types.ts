import { SettingsChangeTransaction } from '@safe-global/store/gateway/AUTO_GENERATED/transactions'
import { Address } from '@/src/types/address'

// TODO: fix it in the @safe-global/store/gateway/AUTO_GENERATED types
export type NormalizedSettingsChangeTransaction = SettingsChangeTransaction & {
  settingsInfo: SettingsChangeTransaction['settingsInfo'] & {
    owner: { value: Address; name: string }
    threshold: number
  }
}
