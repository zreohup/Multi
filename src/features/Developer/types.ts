export type Info = {
  device: {
    brand: string
    deviceName: string
    manufacturer: string
    modelId: string
    modelName: string
    osBuildId: string
    osInternalBuildId: string
    osName: string
    osVersion: string
  }
  application: {
    applicationName: string
    applicationId: string
    applicationVersion: string
    applicationBuildNumber: string
    gatewayUrl: string
    fcmToken: string
  }
}
