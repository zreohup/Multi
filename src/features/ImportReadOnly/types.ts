import { z } from 'zod'
import { formSchema } from '@/src/features/ImportReadOnly/schema'
import { useLazySafesGetOverviewForManyQuery } from '@safe-global/store/gateway/safes'

type LazyQueryResult = ReturnType<typeof useLazySafesGetOverviewForManyQuery>[1]

export type FormValues = z.infer<typeof formSchema> & {
  importedSafeResult?: {
    data: LazyQueryResult['data']
    isFetching: LazyQueryResult['isFetching']
    error: LazyQueryResult['error']
  }
}
