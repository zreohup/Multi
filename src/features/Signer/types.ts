import { z } from 'zod'
import { formSchema } from '@/src/features/Signer/schema'

export type FormValues = z.infer<typeof formSchema>
