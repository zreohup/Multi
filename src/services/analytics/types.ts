import type {
  Transaction,
  TransferTransactionInfo,
  SettingsChangeTransaction,
} from '@safe-global/store/gateway/AUTO_GENERATED/transactions'
export type {
  Transaction,
  TransferTransactionInfo,
  SettingsChangeTransaction,
  CustomTransactionInfo,
  SwapTransferTransactionInfo,
} from '@safe-global/store/gateway/AUTO_GENERATED/transactions'
import type { ANALYTICS_LABELS } from './constants'

/**
 * Firebase Analytics Event Types
 * These event names are passed directly to Firebase Analytics
 */
export enum EventType {
  SCREEN_VIEW = 'screen_view',
  CLICK = 'customClick',
  META = 'metadata',
  SAFE_APP = 'safeApp',
  SAFE_CREATED = 'safe_created',
  SAFE_ACTIVATED = 'safe_activated',
  SAFE_OPENED = 'safe_opened',
  WALLET_CONNECTED = 'wallet_connected',
  TX_CREATED = 'tx_created',
  TX_CONFIRMED = 'tx_confirmed',
  TX_EXECUTED = 'tx_executed',
}

export type EventLabel = string | number | boolean | null

export type AnalyticsEvent = {
  eventName: string
  eventCategory: string
  eventAction: string
  eventLabel?: EventLabel
  chainId?: string
}

export enum AnalyticsUserProperties {
  WALLET_LABEL = 'walletLabel',
  WALLET_ADDRESS = 'walletAddress',
}

// Extract precise types from the source of truth
export type TransactionInfoType = Transaction['txInfo']['type']
export type TransferInfoType = TransferTransactionInfo['transferInfo']['type']
export type SettingsInfoType = SettingsChangeTransaction['settingsInfo']['type']

// Union of all possible analytics labels
export type AnalyticsLabel =
  | (typeof ANALYTICS_LABELS.BASE_TYPES)[TransactionInfoType]
  | (typeof ANALYTICS_LABELS.TRANSFER_TYPES)[TransferInfoType]
  | (typeof ANALYTICS_LABELS.SETTINGS_TYPES)[SettingsInfoType]
  | (typeof ANALYTICS_LABELS.ENHANCED)[keyof typeof ANALYTICS_LABELS.ENHANCED]
