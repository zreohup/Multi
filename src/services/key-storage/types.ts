export type PrivateKeyStorageOptions = {
  requireAuthentication?: boolean
}

export interface IKeyStorageService {
  storePrivateKey(userId: string, privateKey: string, options?: PrivateKeyStorageOptions): Promise<void>
  getPrivateKey(userId: string, options?: PrivateKeyStorageOptions): Promise<string | undefined>
}
