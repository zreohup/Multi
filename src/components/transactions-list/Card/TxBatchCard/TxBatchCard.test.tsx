import { render } from '@/src/tests/test-utils'
import { TxBatchCard } from '.'
import { mockTransferWithInfo } from '@/src/tests/mocks'
import { MultiSend, TransactionInfoType } from '@safe-global/store/gateway/types'

describe('TxBatchCard', () => {
  it('should render the default markup', () => {
    const container = render(
      <TxBatchCard
        onPress={() => null}
        txInfo={
          mockTransferWithInfo({
            type: TransactionInfoType.CUSTOM,
            actionCount: 2,
            to: {
              value: '',
              logoUri: 'http://myAsset.com/asset.png',
              name: 'Gnosis Bridge',
            },
          }) as MultiSend
        }
      />,
    )

    expect(container.getByText('Batch')).toBeTruthy()
    expect(container.getByText('2 actions')).toBeTruthy()
    expect(container).toMatchSnapshot()
  })
})
