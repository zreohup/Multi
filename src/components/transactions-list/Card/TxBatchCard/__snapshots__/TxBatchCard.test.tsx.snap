// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TxBatchCard should render the default markup 1`] = `
<View>
  <View
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "flex-start",
        "backgroundColor": "#FFFFFF",
        "borderBottomLeftRadius": 7,
        "borderBottomRightRadius": 7,
        "borderTopLeftRadius": 7,
        "borderTopRightRadius": 7,
        "flexDirection": "column",
        "flexWrap": "wrap",
        "gap": 12,
        "justifyContent": "flex-start",
        "paddingBottom": 16,
        "paddingLeft": 16,
        "paddingRight": 16,
        "paddingTop": 16,
      }
    }
  >
    <View
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "space-between",
          "width": "100%",
        }
      }
    >
      <View
        style={
          {
            "alignItems": "center",
            "flexDirection": "row",
            "gap": 12,
            "maxWidth": "100%",
          }
        }
      >
        <View
          style={
            {
              "alignItems": "center",
              "borderBottomLeftRadius": 100000,
              "borderBottomRightRadius": 100000,
              "borderTopLeftRadius": 100000,
              "borderTopRightRadius": 100000,
              "flexDirection": "column",
              "height": 40,
              "justifyContent": "center",
              "maxHeight": 40,
              "maxWidth": 40,
              "minHeight": 40,
              "minWidth": 40,
              "overflow": "hidden",
              "paddingBottom": 0,
              "paddingLeft": 0,
              "paddingRight": 0,
              "paddingTop": 0,
              "position": "relative",
              "width": 40,
            }
          }
        >
          <View
            style={
              {
                "bottom": 0,
                "flexDirection": "column",
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
                "zIndex": 1,
              }
            }
          >
            <Image
              accessibilityLabel="Cam"
              fullscreen={true}
              onError={[Function]}
              onLoad={[Function]}
              source={
                {
                  "uri": "http://myAsset.com/asset.png",
                }
              }
              style={
                {
                  "height": 40,
                  "width": 40,
                }
              }
            />
          </View>
          <View
            style={
              {
                "backgroundColor": "#DCDEE0",
                "bottom": 0,
                "flexDirection": "column",
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
                "zIndex": 0,
              }
            }
          >
            <View
              style={
                {
                  "backgroundColor": "#303033",
                  "borderBottomLeftRadius": 100,
                  "borderBottomRightRadius": 100,
                  "borderTopLeftRadius": 100,
                  "borderTopRightRadius": 100,
                  "paddingBottom": 8,
                  "paddingLeft": 8,
                  "paddingRight": 8,
                  "paddingTop": 8,
                }
              }
            >
              <Text
                allowFontScaling={false}
                selectable={false}
                style={
                  [
                    {
                      "color": "#121312",
                      "fontSize": 24,
                    },
                    undefined,
                    {
                      "fontFamily": "SafeIcons",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                    {},
                  ]
                }
              >
                
              </Text>
            </View>
          </View>
        </View>
        <View>
          <View
            style={
              {
                "alignItems": "center",
                "flexDirection": "row",
                "gap": 4,
                "marginBottom": 4,
              }
            }
          >
            <Text
              allowFontScaling={false}
              selectable={false}
              style={
                [
                  {
                    "color": "#A1A3A7",
                    "fontSize": 10,
                  },
                  undefined,
                  {
                    "fontFamily": "SafeIcons",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  {},
                ]
              }
              testID="safe-list-batch-icon"
            >
              
            </Text>
            <Text
              style={
                {
                  "color": "#A1A3A7",
                  "fontFamily": "DM Sans",
                  "fontSize": 13,
                  "marginBottom": 2,
                }
              }
              suppressHighlighting={true}
            >
              Batch
            </Text>
          </View>
          <Text
            style={
              {
                "color": "#121312",
                "fontFamily": "DMSans-SemiBold",
                "fontSize": 14,
              }
            }
            suppressHighlighting={true}
          >
            2 actions
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
